ui/src/pages/execution/Execution.jsx

```diff
@@ -110,17 +112,90 @@ const useStyles = makeStyles({
     alignItems: "center",
     marginRight: 15,
   },
+  dragger: {
+    width: '5px',
+    cursor: 'ew-resize',
+    padding: '4px 0 0',
+    borderTop: '1px solid #ddd',
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    bottom: 0,
+    zIndex: '100',
+    backgroundColor: '#f4f7f9',
+  },
 });
 
-export default function Execution() {
-  const classes = useStyles();
+const actions = {
+  EXPAND_FULL: 1,
+  RESET_EXPAND_FULL: 2,
+  MOUSE_DOWN: 3,
+  MOUSE_UP: 4,
+  MOUSE_MOVE: 5,
+}
+
+const initialState = {
+  drawerWidth: drawerWidth,
+  isDrawerFullScreen: false,
+  isResizing: false,
+  lastDownX: 0,
+  newWidth: {},
+  //drawerVarient: drawerWidth <= 450 ? 'temporary' : 'persistent'
+  drawerVarient: 'persistent'
+};
+
+function reducer(state, action) {
+  switch (action.type) {
+    case actions.EXPAND_FULL:
+      return {...state, isDrawerFullScreen: true, drawerWidth:'100vw'};
+    case actions.RESET_EXPAND_FULL:
+      return {...state, isDrawerFullScreen: false, drawerWidth:drawerWidth, drawerVarient:'persistent'};
+    case actions.MOUSE_DOWN:
+      return {...state, isResizing: true, lastDownX: action.clientX, drawerVarient:'temporary'};
+    case actions.MOUSE_UP:
+      return {...state,  isResizing: false, drawerVarient: state.drawerWidth > maxWindowWidth/2 ? 'temporary' : 'persistent' };
+    case actions.MOUSE_MOVE:
+      return {...state, isDrawerFullScreen: false, drawerWidth: action.offsetRight };
+    default:
+      return state;
+  }
+}
+
+export default function Execution() { 
+  
+  const [state, dispatch] = useReducer(reducer, initialState); 
+  
+  const classes = useStyles(state);
   const match = useRouteMatch();
   const url = `/workflow/${match.params.id}`;
 
   const { data: execution, isFetching, refetch: refresh } = useFetch(url);
 
   const [tabIndex, setTabIndex] = useQueryState("tabIndex", 0);
-  const [selectedTask, setSelectedTask] = useState(null);  
+  const [selectedTask, setSelectedTask] = useState(null); 
+
+  const handleMousedown = (e) => {
+    dispatch({type:actions.MOUSE_DOWN, clientX:e.clientX});
+  };
+
+  const handleMousemove = useCallback((e) => {
+    // we don't want to do anything if we aren't resizing.
+    if (!state.isResizing) {
+      return;
+    }
+
+    let offsetRight =
```

Comment:
nit, since we don't seem to be mutating these variables, would `const` suffice?

Reply:
fixed in other PR #2548